### ISSUE TITLE
Query Parser throws Exception when it shouldn't

### ISSUE THREAD

**Issue by gnat42** [doctrine/doctrine2#5561]
I'm not sure how to have a good title for this however here's the issue. I have a project that is using both MySQL and MSSQL. All queries work fine with MySQL but cause an issue with MSSQL.

DQL: 
```
SELECT MONTH(s.admDate) as AdmissionMonth,COUNT(s.admDx) as admDxCount,s.admDx FROM Entity s
WHERE s....
GROUP BY s.admDx,AdmissionMonth
```

This works perfectly with MySQL. However MSSQL does not support putting a 'variable' in the group by. So for example the above DQL becomes the following SQL:
```
SELECT MONTH(s.admDate) as sclr0 ... GROUP BY s.admDx,sclr0
```
And gets a SQLSTATE exception "Invalid column name 'sclr0'". 

Now I've done some testing, and I can write a SQL query in MySQL and MSSQL as

```
GROUP BY s.admDX,MONTH(s.admDate);
```
and they both work. However Doctrine ORM throws an exception. From Query/Parser.php  (v2.4.8)
```
$this->semanticalError('Cannot group by undefined identification or result variable.');
```

Thoughts?

**Comment by gnat42** [comment]
This seems to be fixed in 2.5.2

**Comment by zeroedin-bill** [comment]
See http://docs.doctrine-project.org/en/latest/reference/native-sql.html#named-native-query
